frontend/src/modules/EditZing/Components/ExportFileIconButton.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import { ReactComponent as CSVExportImg } from '@assets/img/csvexport.svg'
+import { ExportFileIconButtonType } from 'EditZing/Types/ComponentProps'
+import { Button } from '@material-ui/core'
+import { CSV_FILE, DOWNLOAD_ALL, DOWNLOAD_NETIDS } from '@core'
+import { CSVLink } from 'react-csv'
+import { Data } from 'react-csv/components/CommonPropTypes'
+
+// converts contents of the button depending on the type
+export const ExportFileIconButton = ({
+  type,
+  data,
+  downloadData,
+}: ExportFileIconButtonType) => {
+  // fix the typing for data here?
+  const typeToSvg = (
+    type: string,
+    generateData: { (): any[]; (): string | Data }
+  ) => {
+    switch (type) {
+      case CSV_FILE:
+        return (
+          <CSVLink data={generateData()} filename={'groups.csv'}>
+            <CSVExportImg />
+          </CSVLink>
+        )
+    }
+  }
+
+  // gets the onclick behavior
+  const typeToOnClick = () => {
+    switch (type) {
+      case CSV_FILE:
+        // empty function
+        return () => {}
+    }
+  }
+
+  // may need to potentially generate a header as well
+  // generates data on the fly when button gets pressed
```

Comment:
This seems to run when the export button gets pressed, not the icon button to download the file (tested using a console.log lol). Is this intended?

Reply:
Wait I'm not sure what you mean by this. Are you saying that this actually only runs when the export button gets pressed? I have it in the CSVLink component passed as an argument because it returns the data needed for the export, so the CSVLink is "clicked" (by clicking on the button), it would generate data on the fly. It seems to work? I was originally planning to make the CSVLink not wrapped around the button text and then it could be clicked pragmatically when the button itself is clicked, but for some reason TypeScript doesn't want to do the useRef method for pragmatically clicking a link with CSVLink so this was the solution I intended.

Reply:
I'm not sure but when I added a console.log to this function it showed up in the console output when I pressed on the "export" button and had the modal pop up. Think this is because when this component is rendered it runs `typeToSvg` to get the children, which runs `generateData` immediately (thought on the fly meant when the icon button is pressed)